File: backend/app/api/routes/interactions.py
"""Interactions read API -- mentions/replies/DMs.

프론트 mention picker가 답글 보낼 대상 고를 때 사용.
"""

from fastapi import APIRouter, Depends

from app.api.dependencies.auth import get_current_user
from app.api.dependencies.project_access import verify_project_access
from app.core.supabase import supabase

router = APIRouter(prefix="/projects/{project_id}", tags=["interactions"])


@router.get("/interactions")
async def list_interactions(
    project_id: str,
    reply_status: str | None = None,
    limit: int = 30,
    user: dict = Depends(get_current_user),
    _project: dict = Depends(verify_project_access),
):
    """List interactions (mentions/replies/DMs). Default: latest first."""
    query = (
        supabase.table("interactions")
        .select(
            "id, platform, interaction_type, external_id, sender_username, "
            "sender_profile, content, classification, priority, "
            "reply_status, reply_sent_at, draft_reply, detected_at"
        )
        .eq("project_id", project_id)
        .order("detected_at", desc=True)
        .limit(limit)
    )
    if reply_status:
        query = query.eq("reply_status", reply_status)
    return query.execute().data

File: frontend/src/lib/api/interactions.ts
import { apiFetch } from "./client";

export type InteractionType = "mention" | "reply" | "dm" | "comment" | "quote";
export type ReplyStatus =
  | "pending"
  | "draft_ready"
  | "approved"
  | "sent"
  | "ignored"
  | "human_handled";

export type Interaction = {
  id: string;
  platform: string;
  interaction_type: InteractionType | string;
  external_id: string;
  sender_username: string;
  sender_profile: Record<string, unknown>;
  content: string;
  classification: string | null;
  priority: "low" | "medium" | "high" | string;
  reply_status: ReplyStatus | string;
  reply_sent_at: string | null;
  draft_reply: string | null;
  detected_at: string;
};

export async function listInteractions(
  projectId: string,
  opts: { replyStatus?: ReplyStatus; limit?: number } = {},
): Promise<Interaction[]> {
  const params: Record<string, string> = {};
  if (opts.replyStatus) params.reply_status = opts.replyStatus;
  if (opts.limit) params.limit = String(opts.limit);
  return apiFetch<Interaction[]>(
    `/api/projects/${projectId}/interactions`,
    { params },
  );
}
